### PR TITLE
[NOID] Upgrades gradle version

### PR DESCRIPTION
**Pull request by ncordon**
## What
Upgrades gradle to latest 8.x version with:

```
./gradlew wrapper --gradle-version 8.4
```

## Why
Because we were getting errors that seem to be a bug in previous 6.x versions:

```
Could not resolve all dependencies for configuration ':core:testCompileClasspath'.
Problems reading data from Binary store in /private/var/folders/w1/990bdn7j5nb37bbydkqct5gh0000gn/T/gradle12884335207698123056.bin offset 28067 exists? true
```

### CODE REVIEW

**Review by gem-neo4j — APPROVED** [review]
Thanks :D

File: full/build.gradle [hunk 1]
```diff
@@ -18,17 +16,24 @@ spotless {
...
+base {
+    archivesName = "apoc"
+}
 
 jar {
     manifest {
-        attributes 'Implementation-Version': version
+        attributes 'Implementation-Version': archiveVersion
     }
+    archiveClassifier = 'full'
```
Comment (ncordon) [inline full/build.gradle:27]: Make sure this is not breaking the jar classifiers like https://github.com/neo4j/apoc/issues/259#issuecomment-1403779626

**Review by gem-neo4j — APPROVED** [review]
Great, this looks reasonable to me! :D Thank you for doing this! Shall we update 5.x to 8.4 as well? 😁

**Comment by ncordon** [comment]
> Shall we update 5.x to 8.4 as well? 😁

Yep, I'll take care

**Pull request merged**